packages/web3-eth/test/integration/defaults.test.ts

```diff
@@ -0,0 +1,525 @@
+			const eth2 = new Web3Eth({
+				provider: clientUrl,
+				config: {
+					maxListenersWarningThreshold: 4,
+				},
+			});
+			expect(eth2.maxListenersWarningThreshold).toBe(4);
+		});
+		it('defaultNetworkId', async () => {
+			// default
+			expect(web3Eth.defaultNetworkId).toBeNull();
+
+			// after set
+			web3Eth.setConfig({
+				defaultNetworkId: 3,
+			});
+			expect(web3Eth.defaultNetworkId).toBe(3);
+
+			// set by create new instance
+			const eth2 = new Web3Eth({
+				provider: clientUrl,
+				config: {
+					defaultNetworkId: 4,
+				},
+			});
+			expect(eth2.defaultNetworkId).toBe(4);
+			const res = await defaultTransactionBuilder({
+				transaction: {
+					from: '0xEB014f8c8B418Db6b45774c326A0E64C78914dC0',
+					to: '0x3535353535353535353535353535353535353535',
+					value: '0x174876e800',
+					gas: '0x5208',
+				},
+				web3Context: eth2 as Web3Context<any>,
+			});
+			expect(res.networkId).toBe(4);
+		});
+		it('defaultChain', async () => {
+			// default
+			expect(web3Eth.defaultChain).toBe('mainnet');
+
+			// after set
+			web3Eth.setConfig({
+				defaultChain: 'ropsten',
+			});
+			expect(web3Eth.defaultChain).toBe('ropsten');
+
+			// set by create new instance
+			const eth2 = new Web3Eth({
+				provider: clientUrl,
+				config: {
+					defaultChain: 'rinkeby',
+				},
+			});
+			expect(eth2.defaultChain).toBe('rinkeby');
+			const res = await defaultTransactionBuilder({
+				transaction: {
+					from: '0xEB014f8c8B418Db6b45774c326A0E64C78914dC0',
+					to: '0x3535353535353535353535353535353535353535',
+					value: '0x174876e800',
+					gas: '0x5208',
+				},
+				web3Context: eth2 as Web3Context<any>,
+			});
+			expect(res.chain).toBe('rinkeby');
+		});
+		it('defaultHardfork', async () => {
+			// default
+			expect(web3Eth.defaultHardfork).toBe('london');
+
+			// after set
+			web3Eth.setConfig({
+				defaultHardfork: 'dao',
+			});
+			expect(web3Eth.defaultHardfork).toBe('dao');
+
+			// set by create new instance
+			const eth2 = new Web3Eth({
+				provider: clientUrl,
+				config: {
+					defaultHardfork: 'istanbul',
+				},
+			});
+			expect(eth2.defaultHardfork).toBe('istanbul');
+
+			const res = await prepareTransactionForSigning(
+				{
+					from: '0xEB014f8c8B418Db6b45774c326A0E64C78914dC0',
+					to: '0x3535353535353535353535353535353535353535',
+					value: '0x174876e800',
+					gas: '0x5208',
+					gasPrice: '0x4a817c800',
+					data: '0x0',
+					nonce: '0x4',
+					chainId: '0x1',
+					gasLimit: '0x5208',
+				},
+				eth2,
+			);
+			expect(res.common.hardfork()).toBe('istanbul');
+		});
+		it('defaultCommon', () => {
+			// default
+			expect(web3Eth.defaultCommon).toBeNull();
+			const common = {
+				customChain: {
+					name: 'test',
+					networkId: 123,
+					chainId: 1234,
+				},
+				baseChain: 12345,
+				hardfork: 'dao',
+			};
+			// after set
+			web3Eth.setConfig({
+				defaultCommon: common,
+			});
+			expect(web3Eth.defaultCommon).toBe(common);
```

Comment:
we should add tests and verify for defaultCommon being used while signing transaction locally, and also test explicitly provided common obj should override defaultcommon

Reply:
DefaultCommon is not implemented yet.
https://app.zenhub.com/workspaces/web3js-553a980ab285bae538012a40/issues/chainsafe/web3.js/5087

Reply:
I've resolved this conversation.